advanced_source/cpp_custom_ops.rst

```diff
@@ -62,20 +62,64 @@ Using ``cpp_extension`` is as simple as writing the following ``setup.py``:
+  )
 
 If you need to compile CUDA code (for example, ``.cu`` files), then instead use
 `torch.utils.cpp_extension.CUDAExtension <https://pytorch.org/docs/stable/cpp_extension.html#torch.utils.cpp_extension.CUDAExtension>`_.
 Please see `extension-cpp <https://github.com/pytorch/extension-cpp>`_ for an
 example for how this is set up.
 
-Starting with PyTorch 2.6, you can now build a single wheel for multiple CPython
-versions (similar to what you would do for pure python packages). In particular,
-if your custom library adheres to the `CPython Stable Limited API
-<https://docs.python.org/3/c-api/stable.html>`_ or avoids CPython entirely, you
-can build one Python agnostic wheel against a minimum supported CPython version
-through setuptools' ``py_limited_api`` flag, like so:
+The above example represents what we refer to as a CPython agnostic wheel, meaning
+we are building a single wheel that can be run across multiple CPython versions (similar
+to pure Python packages). CPython agnosticism is desirable in minimizing the number of wheels your
+custom library needs to support and release. To achieve this, there are three key lines to note.
+
+The first is the specification of ``Py_LIMITED_API`` in ``extra_compile_args`` to the
+minimum CPython version you would like to support:
+
+.. code-block:: python
+  extra_compile_args={"cxx": ["-DPy_LIMITED_API=0x03090000"]},
+
+Defining the ``Py_LIMITED_API`` flag helps guarantee that the extension is in fact
+only using the `CPython Stable Limited API <https://docs.python.org/3/c-api/stable.html>`_,
+which is a requirement for the building a CPython agnostic wheel. If this requirement
+is not met, it is possible to build a wheel that looks CPython agnostic but will crash,
+or worse, be silently incorrect, in another CPython environment. Take care to avoid
+using unstable CPython APIs, for example APIs from libtorch_python (in particular
+pytorch/python bindings,) and to only use APIs from libtorch (ATen objects, operators
+and the dispatcher). We strongly recommend defining the ``Py_LIMITED_API`` flag to
+ensure the extension is compliant and safe as a CPython agnostic wheel.
+
+The second and third lines inform setuptools that you intend to build a CPython agnostic
+wheel and will influence the naming of the wheel accordingly. It is necessary to specify
+``py_limited_api=True`` as an argument to CppExtension/CUDAExtension and also as an option
+to the ``"bdist_wheel"`` command with the minimal supported CPython version (in this case,
+3.9):
+
+.. code-block:: python
+  setup(name="extension_cpp",
+        ext_modules=[
+            cpp_extension.CppExtension(
+              ...,
+              py_limited_api=True)],  # Build 1 wheel across multiple Python versions
+        ...,
+        options={"bdist_wheel": {"py_limited_api": "cp39"}}  # 3.9 is minimum supported Python version
+  )
+
+This ``setup`` would build one wheel that could be installed across multiple CPython
+versions ``>=3.9``.
```

Comment:
Uh, does the wheel have the right platform tags to reject Python 3.8? Or is the advice to people to always use the smallest version that is not deprecated?

Reply:
the latter hehe 

i haven't checked against 3.8, but it was cuz i chose 3.9 as the min version out there. will include a sentence for that

Reply:
Yeah we need to tell people that the numbers they choose should be the min version pytorch supports